core/coreapi/pin.go

```diff
@@ -67,6 +67,8 @@ func (api *PinAPI) Update(ctx context.Context, from coreiface.Path, to coreiface
 		return err
 	}
 
+	defer api.node.Blockstore.PinLock().Unlock()
```

Comment:
Do we actually need this lock before resolving? (which can take a while in case of `/ipns/` paths, blocking other pin operations)

Reply:
(IIRC this is backed by RWLock, so not that bad of an issue but still)

Reply:
No, but I was trying to match what we do in Pin. I *thought* was also concerned about the fact that we may download the first node and then throw it away but, on closer inspection, that's not a problem. We're keeping it in memory anyways.

I'll fix both of these.